metaflow/cli.py

```diff
@@ -68,20 +72,81 @@ def echo_dev_null(*args, **kwargs):
     pass
 
 
+_animation_thread = None
+_animation_stop = threading.Event()
+_default_spinner = cycle(["-", "\\", "|", "/"])
+
+
+def _animate(get_frame, line, err, kwargs, indent):
+    while not _animation_stop.is_set():
+        frame = get_frame(line)
+        if indent:
+            frame = INDENT + frame
+        terminal_width, _ = shutil.get_terminal_size()
+        frame = frame.ljust(terminal_width)[:terminal_width]
+        click.secho(f"\r{frame}", nl=False, err=err, **kwargs)
+        click.get_text_stream("stderr" if err else "stdout").flush()
+        time.sleep(0.1)
+
+
 def echo_always(line, **kwargs):
+    global _animation_thread, _animation_stop
+
     kwargs["err"] = kwargs.get("err", True)
-    if kwargs.pop("indent", None):
-        line = "\n".join(INDENT + x for x in line.splitlines())
-    if "nl" not in kwargs or kwargs["nl"]:
+    overwrite = kwargs.pop("overwrite", False)
+    animate = kwargs.pop("animate", None)
+    indent = kwargs.pop("indent", False)
+
+    if _animation_thread is not None:
+        _animation_stop.set()
+        _animation_thread.join()
+        _animation_thread = None
+        _animation_stop.clear()
+        click.echo("\r", nl=False, err=kwargs["err"])
+
+    if indent:
+        if animate:
+            # For animated output, we prepend INDENT in the animation function
+            pass
+        else:
+            line = INDENT + line
+
+    animation_kwargs = {
+        "fg": kwargs.get("fg"),
+        "bg": kwargs.get("bg"),
+        "bold": kwargs.get("bold", False),
+        "underline": kwargs.get("underline", False),
+    }
+
+    if animate:
+        if animate is True:
+            get_frame = lambda line: f"{next(_default_spinner)} {line}"
+        else:
+            get_frame = animate
```

Comment:
confused here -- what is animate? It doesn't seem to be a simple bool here so a bit confused.